pkg/controller/kubelet-config/kubelet_config_nodes.go

```diff
@@ -0,0 +1,274 @@
+package kubeletconfig
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/kubelet/config/v1beta1"
+	"reflect"
+	"time"
+
+	"github.com/clarketm/json"
+	"github.com/golang/glog"
+	osev1 "github.com/openshift/api/config/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
+
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	"github.com/openshift/machine-config-operator/pkg/version"
+)
+
+func (ctrl *Controller) nodeWorker() {
+	for ctrl.processNextNodeWorkItem() {
+	}
+}
+
+func (ctrl *Controller) processNextNodeWorkItem() bool {
+	key, quit := ctrl.nodeQueue.Get()
+	if quit {
+		return false
+	}
+	defer ctrl.nodeQueue.Done(key)
+
+	err := ctrl.syncNodeHandler(key.(string))
+	ctrl.handleNodeErr(err, key)
+	return true
+}
+
+func (ctrl *Controller) handleNodeErr(err error, key interface{}) {
+	if err == nil {
+		ctrl.nodeQueue.Forget(key)
+		return
+	}
+
+	if ctrl.nodeQueue.NumRequeues(key) < maxRetries {
+		glog.V(2).Infof("Error syncing node configuration %v: %v", key, err)
+		ctrl.nodeQueue.AddRateLimited(key)
+		return
+	}
+
+	utilruntime.HandleError(err)
+	glog.V(2).Infof("Dropping node config %q out of the queue: %v", key, err)
+	ctrl.nodeQueue.Forget(key)
+	ctrl.nodeQueue.AddAfter(key, 1*time.Minute)
+}
+
+func (ctrl *Controller) syncNodeHandler(key string) error {
+	startTime := time.Now()
+	glog.V(4).Infof("Started syncing node handler %q (%v)", key, startTime)
+	defer func() {
+		glog.V(4).Infof("Finished syncing node handler %q (%v)", key, time.Since(startTime))
+	}()
+
+	// Fetch the Node
+	node, err := ctrl.nodeLister.Get(ctrlcommon.ClusterNodeInstanceName)
+	if errors.IsNotFound(err) {
+		glog.V(2).Infof("Node configuration %v is missing, using default", key)
+		node = createNewDefaultNodeconfig()
+	} else if err != nil {
+		glog.V(2).Infof("%v", err)
+		err := fmt.Errorf("could not fetch Node: %v", err)
+		return err
+	} else {
+		// checking if the Node spec is empty and accordingly returning from here.
+		if reflect.DeepEqual(node.Spec, osev1.NodeSpec{}) {
+			glog.V(2).Info("empty Node resource found")
+			return nil
+		}
+	}
+
+	// Find all MachineConfigPools
+	mcpPools, err := ctrl.mcpLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	for _, pool := range mcpPools {
+		role := pool.Name
+		// the configuration change will be applied only on the worker nodes.
+		if role == string(mcfgv1.MachineConfigPoolMaster) {
+			continue
+		}
+		// Get KubeletConfig
+		managedKey, err := getManagedNodeKey(pool, ctrl.client)
+		if err != nil {
+			return err
+		}
+
+		kc, err := ctrl.client.MachineconfigurationV1().KubeletConfigs().Get(context.TODO(), managedKey, metav1.GetOptions{})
```

Comment:
use argument `key` instead of `managedKey` since it's getting kubeletconfigs.

Reply:
Addressed